lib/awful/spawn.lua

```diff
@@ -370,6 +370,18 @@ function spawn.easy_async(cmd, callback)
     })
 end
 
+--- Call spawn.easy_async with a shell.
+-- This calls `cmd` with `$SHELL -c` (via `awful.util.shell`).
+-- @tparam string cmd The command.
+-- @tab callback Callback function.
```

Comment:
Even if related, `easy_async_with_shell` shouldn't affect or be affected by #1001, I believe. I'll give it a look, anyway.